packages/salesforcedx-vscode-core/test/jest/commands/deployExecutor.test.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { DeployExecutor } from '../../../src/commands/DeployExecutor';
+import { WorkspaceContext } from '../../../src/context/workspaceContext';
+import { SourceTrackingService } from '../../../src/services';
+
+const dummyProjectPath = '/a/project/path';
+jest.mock('@salesforce/source-deploy-retrieve', () => {
+  return {
+    ...jest.requireActual('@salesforce/source-deploy-retrieve'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ComponentSet: jest.fn().mockImplementation(() => {
+      return {
+        deploy: jest.fn().mockImplementation(() => {
+          return { pollStatus: jest.fn() };
+        }),
+        getSourceComponents: jest.fn().mockReturnValue([
+          { name: '1', type: 'ApexClass' },
+          { name: '2', type: 'ApexClass' }
+        ])
+      };
+    })
+  };
+});
+
+jest.mock('@salesforce/salesforcedx-utils-vscode', () => {
+  return {
+    ...jest.requireActual('@salesforce/salesforcedx-utils-vscode'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ChannelService: jest.fn().mockImplementation(() => {
+      return {};
+    }),
+    TelemetryService: { getInstance: jest.fn() },
+    TelemetryBuilder: jest.fn()
+  };
+});
+
+jest.mock('../../../src/messages', () => {
+  return { loadMessageBundle: jest.fn(), nls: { localize: jest.fn() } };
+});
```

Comment:
Just calling it out, but I always try to use the auto mocking if possible just to keep life simple. 

So it'd be something like: 
```
import {messages} from '../../../src/messages';
jest.mock('../../../src/messages');

const mockedMessages = jest.mocked(messages);
```
Then you can just use the fully mocked module like 
```
mockedMessages.nls.localize.mockReturnsValue(jest.fn());
```

Reply:
I always forget about this pattern to get a handle on the mock of the whole dep.  I was able to remove this this time but will keep this pattern in mind!